src/MonitorProfiler/Utilities/OomUtilities.h

```diff
@@ -0,0 +1,8 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#pragma once
+
+#define START_NO_OOM_THROW_REGION try {
+#define END_NO_OOM_THROW_REGION } catch (const std::bad_alloc&) { return E_OUTOFMEMORY; }
+#define ReturnHResultIfOom(exp) START_NO_OOM_THROW_REGION; exp; END_NO_OOM_THROW_REGION;
```

Comment:
The typical pattern we use for these type of macros is "If(Result)Ret" or "If(Result)Ret(Value)"; I would suggest `IfOomRetMem`.